test/CacheTest.php

```diff
@@ -39,14 +39,14 @@ public function resetSkipMethods(array $methods = [])
         $r->setValue(Cache::class, $methods);
     }
 
-    public function testCacheCanAcceptAServerReturningAnArrayOfFunctions()
+    public function testCacheCanAcceptAServerInterfaceReturningAnArrayOfFunctions()
     {
         $functions = [
             'strpos' => 'strpos',
             'substr' => 'substr',
             'strlen' => 'strlen',
         ];
-        $server = $this->prophesize(Server::class);
+        $server = $this->prophesize(ServerInterface::class);
```

Comment:
changes in tests show that we do have BC breaks in this PR, as noted above.

Reply:
Reverted this change.